policy/compiler.go

```diff
@@ -74,17 +74,21 @@ func (c *compiler) compileRule(r *Rule, ruleEnv *cel.Env, iss *cel.Issues) (*com
 	for i, v := range r.Variables() {
 		exprSrc := c.relSource(v.Expression())
 		varAST, exprIss := ruleEnv.CompileSource(exprSrc)
-		if exprIss.Err() == nil {
-			ruleEnv, err = ruleEnv.Extend(cel.Variable(fmt.Sprintf("%s.%s", variablePrefix, v.Name().Value), varAST.OutputType()))
-			if err != nil {
-				iss.ReportErrorAtID(v.Expression().ID, "invalid variable declaration")
-			}
-			compiledVars[i] = &compiledVariable{
-				name: v.name.Value,
-				expr: varAST,
-			}
+		varName := v.Name().Value
+		varType := cel.DynType
```

Comment:
nit: I would add a comment here to explain why we need a DynType by default, it's not obvious.

Reply:
Done.